chapter_3/hello_lib/CMakeLists.txt

```diff
@@ -30,23 +25,45 @@ add_library(hello SHARED)
 # project version * SOVERSION set the compatibility  version for the library to
 # the major number of the version
 set_target_properties(
-  hello
-  PROPERTIES DEBUG_POSTFIX ${CMAKE_DEBUG_POSTFIX}
-             VERSION ${PROJECT_VERSION}
-             SOVERSION ${PROJECT_VERSION_MAJOR})
+    hello
+    PROPERTIES DEBUG_POSTFIX ${CMAKE_DEBUG_POSTFIX}
+               VERSION ${PROJECT_VERSION}
+               SOVERSION ${PROJECT_VERSION_MAJOR}
+)
 
 # add sources to the library target
 target_sources(
-  hello
-  PRIVATE src/hello.cpp src/internal.cpp
-  PUBLIC include/hello/hello.hpp)
+    hello
+    PRIVATE src/hello.cpp src/internal.cpp
+    PUBLIC include/hello/hello.hpp
+)
 
 # define the C++ standard needed to compile this library and make it visible to
 # dependers
 target_compile_features(hello PUBLIC cxx_std_17)
 
 # set the include directories
 target_include_directories(
-  hello
-  PRIVATE src/hello
-  PUBLIC include)
+    hello
+    PRIVATE src/hello
+    PUBLIC include
+)
+
+# An alternative way to create an export header Use either
+# CMAKE_WINDOWS_EXPORT_ALL_SYMBOLS or generate_export_header
+if(MSVC)
+    set(CMAKE_WINDOWS_EXPORT_ALL_SYMBOLS TRUE)
+    set(BUILD_SHARED_LIBS TRUE)
+endif()
```

Comment:
This is not necessary now that the code is using export specifiers.

Reply:
Yes, this is only here to illustrate both ways of exposing symbols. I will should comment it out and leave it in for information purposes.